plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/ide/ext/debugger/client/actions/SelectDebugConfigurationComboBoxAction.java

```diff
@@ -0,0 +1,266 @@
+                                                  EditDebugConfigurationsPresenter editConfigurationsPresenter,
+                                                  DebugConfigurationsManager debugConfigurationsManager) {
+        super(Collections.singletonList(PROJECT_PERSPECTIVE_ID),
+              locale.selectConfigurationActionText(),
+              locale.selectConfigurationActionDescription(),
+              null, null);
+        this.locale = locale;
+        this.resources = resources;
+        this.actionManager = actionManager;
+
+        this.dropDownListFactory = dropDownListFactory;
+        this.debugConfigurationsManager = debugConfigurationsManager;
+        this.dropDownHeaderWidget = dropDownListFactory.createList(GROUP_DEBUG_CONFIGURATIONS_LIST);
+
+        configurations = new LinkedList<>();
+
+        eventBus.addHandler(WsAgentStateEvent.TYPE, this);
+        editConfigurationsPresenter.addConfigurationsChangedListener(this);
+
+        configurationActions = new DefaultActionGroup(GROUP_DEBUG_CONFIGURATIONS, false, actionManager);
+        actionManager.registerAction(GROUP_DEBUG_CONFIGURATIONS, configurationActions);
+
+        dropDownHeaderWidget.setDelegate(this);
+    }
+
+    @Override
+    public void updateInPerspective(@NotNull ActionEvent event) {
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+    }
+
+    @Override
+    public Widget createCustomComponent(Presentation presentation) {
+        FlowPanel customComponentHeader = new FlowPanel();
+        FlowPanel debugIconPanel = new FlowPanel();
+
+        customComponentHeader.setStyleName(resources.getCss().selectConfigurationBox());
+        debugIconPanel.setStyleName(resources.getCss().selectConfigurationsBoxIconPanel());
+        debugIconPanel.add(new SVGImage(resources.debugIcon()));
+        customComponentHeader.add(debugIconPanel);
+        customComponentHeader.add((Widget)dropDownHeaderWidget);
+
+        return customComponentHeader;
+    }
+
+    @Override
+    public void onSelect() {
+    }
+
+    /** Returns the selected debug configuration. */
+    @Nullable
+    public DebugConfiguration getSelectedConfiguration() {
+        if (configurations.isEmpty()) {
+            return null;
+        }
+
+        final String selectedConfigurationName = dropDownHeaderWidget.getSelectedElementName();
+
+        for (DebugConfiguration configuration : configurations) {
+            if (configuration.getName().equals(selectedConfigurationName)) {
+                return configuration;
+            }
+        }
+        return null;
+    }
+
+    public void setSelectedConfiguration(DebugConfiguration configuration) {
+        dropDownHeaderWidget.selectElement(configuration.getName());
+    }
+
+    /** Returns configuration by it's name. */
+    @Nullable
+    public DebugConfiguration getConfigurationByName(String name) {
+        if (configurations.isEmpty()) {
```

Comment:
"Edit Debug Configurations" should be bind to "Alt+Shift+F9"
"Debug" should be find to "Shift+F9"